### PR TITLE
rework libeinfo

### PR DESCRIPTION
**Pull request by xxc3nsoredxx**
Rework how libeinfo does curses stuff. This is not by any means ready to merge as-is. For example, OpenRC core has not been changed where needed (places where `ecolor` is called are the most obvious ones). There also isn't a non-curses fallback yet, but that should be easy to add once everything else is satisfactory. Not to mention squashing/rewording commits.

This is just to get some feedback on what I currently have.

See also: #619

### CODE REVIEW

**Comment by xxc3nsoredxx** [comment]
Squashed a few of the commits/cleaned up the messages.

**Comment by williamh** [comment]
We don't need command line equivalents for ewarnx and eerrorx.
In a script, you can use ewarn then exit or do the same with eerror.

**Comment by williamh** [comment]
Please rebase on master.

I also see a couple of changes.

On line 50 of einfo.h:

```
__attribute__((__deprecated__("this shit is gone yo")))
```

What does this apply to? Also, can you fix the message to be something like "this is no longer used"?

Please update if statements that just have one statement in the if or else blocks to not use curly braces around the statement:

```
if (foo) {
    bar;
}
```

should be:

```
if (foo)
    bar;
```


Can you also remove the vim mode lines?

Thanks,

William

**Comment by xxc3nsoredxx** [comment]
I can do the single-line `if` thing, sure. The `__deprecated__` message is currently just a placeholder to alert me to things which use the `ecolor()` function in OpenRC and will be changed before I un-draft the PR. Same with removing the mode line.

File: src/libeinfo/einfo.h [hunk 1]
```diff
@@ -47,6 +47,7 @@ typedef enum
 } ECOLOR;
 
 /*! @brief Returns the ASCII code for the color */
+__attribute__((__deprecated__("this shit is gone yo")))
```
Comment (vapier) [inline src/libeinfo/einfo.h:50]: sorry, but we can't accept things like this.  ignoring the puerile nature, it's not helpful or clear to anyone who hits/sees this warning.

**Review by vapier — CHANGES_REQUESTED** [review]
there's a lot of commits in this stack.  will prob need to pull out some of the smaller ones independently.

**Comment by vapier** [comment]
> there's a lot of commits in this stack. will prob need to pull out some of the smaller ones independently.

ok, i flipped through the patch series now, and it doesn't make sense to pull things out.  this is a PR showing progress of implementing a new libeinfo library and would make sense squashing all into one.

i'm inclined to go with my half-baked idea that i mentioned in the open issue.  since we already support building openrc w/out termcap (curses) support, we should lean into that: delete support for termcap entirely and just require an ANSI compliant terminal.

File: src/libeinfo/einfo.h (hunk 1 above)
Comment (xxc3nsoredxx) [reply src/libeinfo/einfo.h:50]: It was merely meant as a message while I was working and not for the final cleaned up/squashed commits.

**Comment by xxc3nsoredxx** [comment]
I can close this PR if you instead want to just drop curses and require an ANSI terminal.

**Comment by vapier** [comment]
you want to send a PR to delete curses support ?

**Comment by xxc3nsoredxx** [comment]
Sure, I think I can do that.

**Pull request closed without merging**